dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -182,27 +182,25 @@ public sealed class SKFunction : ISKFunction, IDisposable
                 var completionResults = await client.GetCompletionsAsync(renderedPrompt, requestSettings, cancellationToken).ConfigureAwait(false);
                 string completion = await GetCompletionsResultContentAsync(completionResults, cancellationToken).ConfigureAwait(false);
 
-                // Update the result with the completion
-                context.Variables.Update(completion);
-
-                context.ModelResults = completionResults.Select(c => c.ModelResult).ToArray();
+                // Create a copy of the SKContext, update the result with the completion
+                var contextResult = new DefaultSKContext(context);
+                contextResult.Variables.Update(completion);
+                return contextResult;
             }
             catch (AIException ex)
             {
                 const string Message = "Something went wrong while rendering the semantic function" +
                                        " or while executing the text completion. Function: {0}.{1}. Error: {2}. Details: {3}";
                 logger?.LogError(ex, Message, skillName, functionName, ex.Message, ex.Detail);
-                context.Fail(ex.Message, ex);
+                return DefaultSKContext.FromException(ex);
```

Comment:
The returned context doesn't need to preserve things like input variables?

Reply:
Unclear. This is a stepping stone change to one where we replace the return type of all kernel operations, SKContext, with something specifically resulty, like SKResult.

Good point though, I'll look at keeping as much in the context as possible (for now), even in a Fail condition.